server/manager/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -587,8 +586,9 @@ TabletGoalState getTableGoalState(KeyExtent extent) {
   TabletGoalState getGoalState(TabletLocationState tls, MergeInfo mergeInfo) {
     KeyExtent extent = tls.extent;
     // Shutting down?
-    TabletGoalState state = getSystemGoalState(tls);
+    TabletGoalState state = getSystemGoalState(extent);
     if (state == TabletGoalState.HOSTED) {
+      // TODO shouldn't this be in the upgrade code? this seems very inefficient
```

Comment:
I feel like this check should be moved to the upgrade code. Perhaps it was missed in the upgrade code wrangling?  Checking on every single tablet assignment seems very inefficient.